### PR TITLE
feat(emails): Add client js support for adding more emails

### PR DESCRIPTION
**Pull request by vbudhram**
Connects with https://github.com/mozilla/fxa-content-server/pull/4801

### CODE REVIEW

File: client/FxAccountClient.js [hunk 1]
```diff
@@ -1203,6 +1203,67 @@ define([
...
+   * @method getEmails
+   * @param {String} sessionToken SessionToken obtained from signIn
+   */
+  FxAccountClient.prototype.getEmails = function (sessionToken) {
+    required(sessionToken, 'sessionToken');
+
+    var request = this.request;
+    return hawkCredentials(sessionToken, 'sessionToken',  HKDF_SIZE)
+      .then(function(creds) {
+        return request.send('/recovery_email/emails', 'GET', creds, {}, {});
```
Comment (vladikoff) [inline client/FxAccountClient.js]: from mtg: will need to double check this

Comment (vladikoff) [inline client/FxAccountClient.js]: clean up `{}, {}`

File: client/FxAccountClient.js [hunk 2]
```diff
@@ -1203,6 +1203,67 @@ define([
   };
 
   /**
+   * Get all the emails associated with user that belongs to the
+   * session.
+   *
+   * @method getEmails
+   * @param {String} sessionToken SessionToken obtained from signIn
+   */
+  FxAccountClient.prototype.getEmails = function (sessionToken) {
```
Comment (vladikoff) [inline client/FxAccountClient.js]: `getEmails` -> `emails`? mayyyybee

File: client/FxAccountClient.js [hunk 3]
```diff
@@ -1227,6 +1235,67 @@ define([
   };
 
   /**
+   * Get all the recovery emails associated with user that belongs to the
```
Comment (shane-tomlinson) [inline client/FxAccountClient.js]: I'd really like to learn the history of the name `recovery emails`...

File: client/FxAccountClient.js [hunk 4]
```diff
@@ -329,6 +333,10 @@ define([
     required(sessionToken, 'sessionToken');
 
     if (options) {
+      if (options.email) {
```
Comment (shane-tomlinson) [inline client/FxAccountClient.js]: Can you add `email` to the `options` docs above?

File: client/FxAccountClient.js [hunk 5]
```diff
@@ -283,6 +283,10 @@ define([
       if (options.reminder) {
         data.reminder = options.reminder;
       }
+
+      if (options.type) {
```
Comment (shane-tomlinson) [inline client/FxAccountClient.js]: Can you add `type` to the `options` docs above?

File: client/FxAccountClient.js [hunk 6]
```diff
@@ -1227,6 +1235,67 @@ define([
...
+  };
+
+  /**
+   * Remove the email address from the user account
+   *
+   * @method deleteEmail
+   * @param {String} sessionToken SessionToken obtained from signIn
+   * @param {String} email email to be removed
+   */
+  FxAccountClient.prototype.deleteEmail = function (sessionToken, email) {
```
Comment (shane-tomlinson) [inline client/FxAccountClient.js]: rather than `delete`, it seems `destroy` is the common verb in this module.

File: client/FxAccountClient.js [hunk 7]
```diff
@@ -1227,6 +1235,67 @@ define([
...
+    required(sessionToken, 'email');
+
+    var data = {
+      email: email
+    };
+
+    var request = this.request;
+    return hawkCredentials(sessionToken, 'sessionToken',  HKDF_SIZE)
+      .then(function(creds) {
+        return request.send('/recovery_email/destroy', 'POST', creds, data, {});
```
Comment (shane-tomlinson) [inline client/FxAccountClient.js]: What is the 5th option to `send`, the empty `{}` - other POST requests don't send it, is it necessary?

Comment (vladikoff) [inline client/FxAccountClient.js]: nope!

**Comment by vbudhram** [comment]
Thanks for feedback @shane-tomlinson, made the updates you requested.

File: client/FxAccountClient.js [hunk 8]
```diff
@@ -1227,6 +1239,66 @@ define([
...
+    return hawkCredentials(sessionToken, 'sessionToken',  HKDF_SIZE)
+      .then(function(creds) {
+        return request.send('/recovery_emails', 'GET', creds);
+      });
+  };
+
+  /**
+   * Create a new recovery email for the signed in account.
+   *
+   * @method recoveryEmailAdd
```
Comment (vladikoff) [inline client/FxAccountClient.js]: `@method` mismatch vs `recoveryEmailCreate` below

**Pull request merged**